crates/ruma-client-api/src/sync/sliding_sync_events.rs

```diff
@@ -0,0 +1,310 @@
+    /// Sticky. Flag which only returns rooms present (or not) in the DM section of account data.
+    /// If unset, both DM rooms and non-DM rooms are returned. If false, only non-DM rooms
+    /// are returned. If true, only DM rooms are returned.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub is_dm: Option<bool>,
+
+    /// Sticky. A list of spaces which target rooms must be a part of. For every invited/joined
+    /// room for this user, ensure that there is a parent space event which is in this list. If
+    /// unset, all rooms are included. Servers MUST NOT navigate subspaces. It is up to the
+    /// client to give a complete list of spaces to navigate. Only rooms directly in these
+    /// spaces will be returned.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub spaces: Option<Vec<String>>,
+
+    /// Sticky. Flag which only returns rooms which have an `m.room.encryption` state event. If
+    /// unset, both encrypted and unencrypted rooms are returned. If false, only unencrypted
+    /// rooms are returned. If true, only encrypted rooms are returned.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub is_encrypted: Option<bool>,
+
+    /// Sticky. Flag which only returns rooms the user is currently invited to. If unset, both
+    /// invited and joined rooms are returned. If false, no invited rooms are returned. If
+    /// true, only invited rooms are returned.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub is_invite: Option<bool>,
+
+    /// Flag which only returns rooms which have an `m.room.tombstone` state event. If unset,
+    /// both tombstoned and un-tombstoned rooms are returned. If false, only un-tombstoned rooms
+    /// are returned. If true, only tombstoned rooms are returned.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub is_tombstoned: Option<bool>,
+
+    /// If specified, only rooms where the `m.room.create` event has a `type` matching one
+    /// of the strings in this array will be returned. If this field is unset, all rooms are
+    /// returned regardless of type. This can be used to get the initial set of spaces for an
+    /// account.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub room_types: Option<Vec<String>>,
+
+    /// Same as "room_types" but inverted. This can be used to filter out spaces from the room
+    /// list.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub not_room_types: Option<Vec<String>>,
+
+    /// Filter the room name. Case-insensitive partial matching e.g 'foo' matches 'abFooab'.
+    /// The term 'like' is inspired by SQL 'LIKE', and the text here is similar to '%foo%'.
+    pub room_name_like: Option<String>,
+}
+
+/// Sliding Sync Request for each list
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct SyncRequestList {
+    /// The ranges of rooms we're interested in
```

Comment:
Docs should be terminated with a period.

```suggestion
    /// The ranges of rooms we're interested in.
```